pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LinuxBrokerHostUsageImpl.java

```diff
@@ -231,8 +232,15 @@ private boolean isPhysicalNic(Path path) {
                 Files.readAllBytes(path.resolve("speed"));
                 return true;
             } catch (Exception e) {
-                // wireless nics don't report speed, ignore them.
-                return false;
+                // In some cases, VMs in EC2 won't have the speed reported on the NIC and will give a read-error.
+                // Check the type to make sure it's ethernet (type "1")
+                try {
+                    String type = new String(Files.readAllBytes(path.resolve("type")), StandardCharsets.UTF_8).trim();
+                    return Integer.parseInt(type) == 1;
```

Comment:
@merlimat - by returning true here, in some cases, don't we also need to update the logic in getNicSpeedPath? We're seeing the new error here https://github.com/apache/pulsar/pull/14340 because this class is returning true in a new case.

Reply:
> When that happens, we're ignoring that NIC and it causes that we cannot even manually override the NIC capacity in broker.conf

I see now that the point was to make it so it could be overridden.